src/manifolds/StiefelFrechet.jl

```diff
@@ -0,0 +1,312 @@
+_g_rank_cutoff = 1e-14  # cut off of eigenvalue to determine rank of span of two orthogonal basis
```

Comment:
These first seven lines contain some “magic numbers” that not only have names I do not understand, but even more it would be better if a user could set these. Can you put them into their corresponding functions as keyword arguments please?

Reply:
they actually could be overwritten in lbfgs_options - I changed the codes to make it clearer. In the docstring I explained what the variable means 

102    if !isnothing(lbfgs_options)
103        complementary_rank_cutoff = get(lbfgs_options, "complementary_rank_cutoff", 1e-14)
104        corrections = get(lbfgs_options, "corrections", 10)
105        c1 = get(lbfgs_options, "c1", 1e-4)
106        c2 = get(lbfgs_options, "c2", 0.9)
107        max_ls = get(lbfgs_options, "max_ls", 25)
108        max_fun_evals = get(lbfgs_options, "max_fun_evals", Int(ceil(max_itr * 1.3)))
109    else
110        complementary_rank_cutoff = 1e-14
111        corrections = 10
112        c1 = 1e-4
113        c2 = 0.9
114        max_ls = 25
115        max_fun_evals = Int(ceil(max_itr * 1.3))
116    end

From the doc string (sorry the HTML paste doesnt work well here - but should be visible on the website).
lbfgs_options is a dictionary of technical parameters, isolated to a separate group as we do not expect users to modify them. The relevant keys are

    complementary_rank_cutoffcomplementary\_rank\_cutoffcomplementary_rank_cutoff (default = 1e-14): cut off of eigenvalue to determine the rank of the complementary basis QQQ. We use SVD to determine the size of the complement basis QQQ of YYY in span(Y,W)span(Y, W)span(Y,W). We keep columns of QQQ corresponding to eigenvalues higher than this cutoff.

    The remaining parameters are related to the L-BFGS minimizer minimize:
        correctionscorrectionscorrections (default = 10): default number of stored history in L-BFGS.
        c1c1c1 (default = 1e-4) default c1, c2 in the line search.
        c2c2c2 (default = 0.9).
        max_lsmax\_lsmax_ls (default = 25) max iterations for the line search.
        maxfunevalsmax_fun_evalsmaxf​une​vals (default = Int(ceil(max_itr * 1.3))) max function evaluations